### PR TITLE
feat: endpoint exposed for changing online status of locations

### PR DESCRIPTION
**Pull request by appmdev**
# Description

An endpoind was created in Central API that allow to change isOnline status and DateTime of change of a location (Edge/Cloud) which will be used by Heartbeat.
Alongside a worker job was created using quartz to check every minute if each locations was uptaded with "isOnline status = true" in less then 3 minutes otherwise it changes "isOnline status = false";

Fixes # allow to change isOnline status of location(edge/cloud); monitor online status of locations(edge/cloud)

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## What has been changed?

- Feature: Heartbeat can change isOnline status of Locations(edge/cloud)

# How Has This Been Tested?

The test checked if for givven correct id and location, would change isOnline status of location, alongside with wrong inputs. Futhe more, it checked if worker job would change isOnline status to false of locations which was online and not updated for more then three minutes.

- [x] Test A: the correct id and location was given, to check isOnline status; the result was as expected; it returned correct isOnline value for every requests.
- [x] Test B: the id and location was given wrong in any mix and macth cases; the output was "not found" or "bad request"
- [x] Test C: the worker job was checked if would change status of locations which was online and not updated for more then three minutes; the output was successful, the locations with "isOnline=true" which isOnline status was updated less then three minutes was unchanged, and for rest of them isOnline status was changed to false;

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/DataAccess/Repositories/Redis/RedisEdgeRepository.cs [hunk 1]
```diff
@@ -270,5 +270,54 @@ public async Task<(bool, EdgeModel?)> CheckIfIdExists(string id)
                 return (false, matchedEdge);
             }
         }
+
+        /// <summary>
+        /// Return online status and LastUpdatedTime of edge by cloud Id.
+        /// </summary>
+        /// <param name="edgeId"></param>
+        /// <returns>Return bool of isOnline and dateTime of lastUpdatedTime of the edgeModel</returns>
+        [Obsolete]
```
Comment (Artonus) [inline src/DataAccess/Repositories/Redis/RedisEdgeRepository.cs:279]: Should this function be `Obsolete`? This tag means that the function is unused and will shortly be removed

File: src/Models/Domain/CloudEdgeStatusResponse.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using Middleware.Models.Enums;
+
+namespace Middleware.Models.Domain;
+//internal class CloudEdgeStatus
+//{
+//}
+
+public class CloudEdgeStatusResponse
```
Comment (Artonus) [inline src/Models/Domain/CloudEdgeStatusResponse.cs:13]: @appmdev can you move these two classes to the `CentralApi.Contracts` project? They are used only for the communication with CentralApi, so their place should be there :)

**Review by Artonus — CHANGES_REQUESTED** [review]
Vald, this looks good for me, but I would ask you to correct 2 things:

- Please add missing `await` operators to the async functions you are using. It is marked with the comments in the PR
- Please correct the tests in `test/CentralApi.Tests.Unit/Controller/LocationsControllerTests.cs` so they will be passing. You just have to add 2 new parameters to the constructor and that should be fine.

The rest is ok, some suggestions that could improve overall code organization, but they're not critical for the code working correctly. 
The 2 points above could cause us deadlocks because of the missing `await` and not passing tests will not allow merging your changes and then building the app.

File: src/CentralApi/Jobs/UpdateOnlineStatusJob.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        _logger = logger;
+        _cloudRepository = cloudRepository;
+        _edgeRepository = edgeRepository;
+    }
+
+    protected override async Task ExecuteJobAsync(IJobExecutionContext context)
+    {
+        try
+        {
```
[the remark this answers is not in the document]
Comment (appmdev) [reply src/CentralApi/Jobs/UpdateOnlineStatusJob.cs:25]: each call is intependent, can not treat as one unit.

**Review by Artonus — APPROVED** [review]

**Pull request merged**